float32 posx
float32 posy
bool rightWall
bool robot1
---
bool success
bool robot1
---
float32 progress
bool robot1